src/Approvals/SFA.DAS.Approvals/Application/Cohorts/Queries/GetCohortDetails/GetCohortDetailsQuery.cs

```diff
@@ -22,12 +22,35 @@ public class GetCohortDetailsQuery : IRequest<GetCohortDetailsQueryResult>
 
     public class GetCohortDetailsQueryResult
     {
-        public string ProviderName { get; set; }
+        public long CohortId { get; set; }
+        public string CohortReference { get; set; }
+        public long AccountId { get; set; }
+        public long AccountLegalEntityId { get; set; }
         public string LegalEntityName { get; set; }
         public bool HasNoDeclaredStandards { get; set; }
+        public string ProviderName { get; set; }
+        public long? ProviderId { get; set; }
+        public bool IsFundedByTransfer { get; set; }
+        public long? TransferSenderId { get; set; }
+        public int? PledgeApplicationId { get; set; }
+        public Party WithParty { get; set; }
+        public string LatestMessageCreatedByEmployer { get; set; }
+        public string LatestMessageCreatedByProvider { get; set; }
+        public bool IsApprovedByEmployer { get; set; }
+        public bool IsApprovedByProvider { get; set; }
+        public bool IsCompleteForEmployer { get; set; }
+        public bool IsCompleteForProvider { get; set; }
+        public ApprenticeshipEmployerType LevyStatus { get; set; }
+        public long? ChangeOfPartyRequestId { get; set; }
+        public bool IsLinkedToChangeOfPartyRequest { get; set; }
+        public TransferApprovalStatus? TransferApprovalStatus { get; set; }
+        public LastAction LastAction { get; set; }
+        public bool ApprenticeEmailIsRequired { get; set; }
         public bool HasUnavailableFlexiJobAgencyDeliveryModel { get; set; }
         public IEnumerable<string> InvalidProviderCourseCodes { get; set; }
-
+        public IEnumerable<DraftApprenticeship> DraftApprenticeships { get; set; }
```

Comment:
I don't love this - it means that the query result is partly defined by the definition of an inner api type. The query result's definition should be entirely self-contained, even if that seems like a lot of mapping overhead.

But you can see the problem: if the inner api's definition changes, the query result's definition will also change, and that could have unforseen consequences. You can usually get away with this, but if you are _also_ exposing query results directly to the outside world then this could be a breaking change to clients of this api. (If you are having to change the UI layer because the inner api has changed, then something is badly wrong).

Reply:
Will write a sub type in the response object, but think we might get duplication coverage issues. And I hate the layers of mappings we are introducing. I much prefer the passthrough mechanisms when it's a call just to your own inner API.